### PR TITLE
Persist-cache: add negative obj cache

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/CacheBackend.java [hunk 1]
```diff
@@ -44,6 +52,12 @@ public interface CacheBackend {
   /** Adds the given object only to the local cache, does not send a cache-invalidation message. */
   void putLocal(@Nonnull String repositoryId, @Nonnull Obj obj);
 
+  /**
+   * Record the "not found" sentinel for the given {@link ObjId id} and {@link ObjType type}.
+   * Behaves like {@link #remove(String, ObjId)}, if {@code type} is {@code null}.
+   */
+  void putNegative(@Nonnull String repositoryId, @Nonnull ObjId id, @Nonnull ObjType type);
```
Comment (dimas-b) [inline versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/CacheBackend.java:59]: nit: why not keep it close (line-wise) to the other `putNegative` method?

File: versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/ObjCache.java [hunk 2]
```diff
@@ -55,4 +69,22 @@ public interface ObjCache {
   void putReferenceLocal(@Nonnull Reference r);
 
   void putNegative(@Nonnull String name);
+
+  /**
+   * Special sentinel object instance to indicate that an object has been marked as "not found" via
+   * {@link #putNegative(ObjId, ObjType)}, returned from {@link #get(ObjId)}. This object is only
+   * for cache-internal purposes.
+   */
+  Obj NOT_FOUND_OBJ_SENTINEL =
```
Comment (dimas-b) [inline versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/ObjCache.java:78]: Why do we keep this constant here, but `NON_EXISTENT_REFERENCE_SENTINEL` in `CacheBackend`?

**Review by dimas-b — COMMENTED** [review]
LGTM overall, just a couple of minor comments about code locations :)

File: versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/ObjCache.java (hunk 2 above)
Comment (snazy) [reply versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/ObjCache.java:78]: moved

File: versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/CacheBackend.java (hunk 1 above)
Comment (snazy) [reply versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/CacheBackend.java:59]: it's reference related - renamed it

**Review by dimas-b — APPROVED** [review]

**Pull request merged**